### ISSUE TITLE
Docs overhaul

### ISSUE THREAD

**Issue by krzkaczor** [earl-js/earl#66]
TODO:

- [ ] upgrade docozaurus
- [ ] add searchbox
- [ ] add example to the main page

**Comment by hasparus** [comment]
> add searchbox

https://docusaurus.io/docs/search#using-algolia-docsearch

Shall we apply for Algolia DocSearch? (It went pretty smoothly for Theme UI and the search experience is top notch.)

**Comment by krzkaczor** [comment]
> Shall we apply for Algolia DocSearch? (It went pretty smoothly for Theme UI and the search experience is top notch.)

Yes, that was my idea. The problem is that our docosaurus instance is unupgradable at the moment because I modified it to look better 😅  I hate the original look and I tweaked some elements so it looks more like gitbook. It would be great to redo it as some kind of theme...

**Comment by hasparus** [comment]
Oh we could use DocSearch even without Docusaurus, so these two don't conflict. I just linked the docusaurus docs as I had them already open.

I like Supabase's approach

- They have a nice introduction page as docusaurus home page: https://supabase.com/docs 
- and a custom landing page with all the cool stuff to sell the idea: https://supabase.com/

**Comment by krzkaczor** [comment]
More ideas:
* real-world usage guide,
* more examples,
* grammar check,

**Comment by sz-piotr** [comment]
Closing as sub issues were created
